### PR TITLE
Adding ADAGE model to tsne visualization

### PR DESCRIPTION
**Pull request by gwaygenomics**
Quick check if ADAGE (`z=100`) is learning large sources of variation in the data.

### CODE REVIEW

File: scripts/viz/tsne_viz.R [hunk 1]
```diff
@@ -1,25 +1,47 @@
 # Pan-Cancer Variational Autoencoder
 # Gregory Way 2017
 #
-# Visualize t-SNE plots of VAE and raw RNAseq data features
+# Visualize t-SNE plots of VAE, ADAGE, and raw RNAseq data features
```
Comment (ajlee21) [inline scripts/viz/tsne_viz.R]: Add version of R used since there is some versioning issues

File: scripts/viz/tsne_viz.R [hunk 2]
```diff
@@ -1,25 +1,47 @@
...
 tsne_vae_df <- readr::read_tsv(tsne_vae_file)
+tsne_adage_df <- readr::read_tsv(tsne_adage_file)
 tsne_rnaseq_df <- readr::read_tsv(tsne_rnaseq_file)
 
 tsne_vae_df <- dplyr::inner_join(tsne_vae_df, clinical_df,
                                  by = c("tcga_id" = "sample_id"))
+tsne_adage_df <- dplyr::inner_join(tsne_adage_df, clinical_df,
+                                 by = c("tcga_id" = "sample_id"))
 tsne_rnaseq_df <- dplyr::inner_join(tsne_rnaseq_df, clinical_df,
                                     by = c("tcga_id" = "sample_id"))
```
Comment (ajlee21) [inline scripts/viz/tsne_viz.R]: Looks like you're applying the same commands several times, not sure if there is a "sleeker" way to do this in R like there is in python:

data_folder = 'C:/Users/alexj/Documents/UPenn/GCB537/Ex1prog/'
file_motifs = [data_folder + 'Output_motif%d_sequences_modified.txt' % (i+1) for i in xrange(n_data_files)]

Just a question

File: scripts/viz/tsne_viz.R [hunk 3]
```diff
@@ -1,25 +1,47 @@
...
 
 library(dplyr)
 library(ggplot2)
 
+plot_tsne <- function(tsne_df) {
+  # Function to output tsne visualizations colored by cancer type for the given
+  # input data (compressed by either Tybalt or ADAGE or raw RNAseq data)
+  p <- ggplot(tsne_df, aes(x = `1`, y = `2`, color = acronym)) +
+    geom_point(size = 0.001) +
+    scale_colour_manual(limits = tcga_colors$`Study Abbreviation`,
```
Comment (ajlee21) [inline scripts/viz/tsne_viz.R]: I was personally a bit confused having the function defined before tcga_colors but I think that may be a personal style choice.

File: scripts/viz/tsne_viz.R [hunk 4]
```diff
@@ -1,25 +1,47 @@
...
+          legend.text = element_text(size = 5),
+          legend.key.size = unit(0.5, "line")) +
+    guides(colour = guide_legend(override.aes = list(size = 0.1),
+                                 title = 'Cancer-Type'))
+  return(p)
+}
+
 # Load encodings file with matched clinical data to subset
 clinical_file <- file.path("data", "tybalt_features_with_clinical.tsv")
 clinical_df <- readr::read_tsv(clinical_file)
```
Comment (ajlee21) [inline scripts/viz/tsne_viz.R]: Does the clinical_df need to include the gene expression data in addition to the sample id, or can this be the sample id's and cancer type and then you can apply "inner join" on the sample id?

**Comment by ajlee21** [comment]
Here are my comments Greg.  This was really helpful for me to review as I start to work on my t-SNE for GTEx!

File: scripts/viz/tsne_viz.R (hunk 3 above)
Comment (gwaygenomics) [reply scripts/viz/tsne_viz.R]: you're right - not a style choice but a departure from proper R. I will fix!

File: scripts/viz/tsne_viz.R (hunk 4 above)
Comment (gwaygenomics) [reply scripts/viz/tsne_viz.R]: yeah, that is definitely an option - i had previously saved this file for use in a different script that required it, so I figured I would use it here!

File: scripts/viz/tsne_viz.R (hunk 1 above)
Comment (gwaygenomics) [reply scripts/viz/tsne_viz.R]: Great point. I realize now that I don't have a conda environment for R scripts!! I will need to add one, but it is beyond the scope of this pull request.

**Comment by gwaygenomics** [comment]
Thanks @ajlee21 - I believe I addressed all of your concerns in 05c2b5b

File: scripts/viz/tsne_viz.R [hunk 5]
```diff
@@ -1,25 +1,47 @@
 # Pan-Cancer Variational Autoencoder
 # Gregory Way 2017
 #
-# Visualize t-SNE plots of VAE and raw RNAseq data features
+# Visualize t-SNE plots of VAE, ADAGE, and raw RNAseq data features
 
 library(dplyr)
 library(ggplot2)
```
Comment (ajlee21) [inline scripts/viz/tsne_viz.R]: One last comment: Think you need to add library(readr) here

Comment (gwaygenomics) [reply scripts/viz/tsne_viz.R]: In R, the [`double colon notation`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/ns-dblcolon.html) extracts the function from the namespace directly, so there is no need to load the library in this case.

Comment (ajlee21) [reply scripts/viz/tsne_viz.R]: I had issues running the script without loading the package first.  Maybes its a versioning issue, which in that case your env script will address

Comment (gwaygenomics) [reply scripts/viz/tsne_viz.R]: hmm, interesting. I will cross reference this note in the issue

Comment (gwaygenomics) [reply scripts/viz/tsne_viz.R]: I also just added loading the library in caaa13f so that others don't run into this issue before I address #115

**Pull request merged**